### PR TITLE
add custom packages for Full VM && Micro VM

### PR DESCRIPTION
**Pull request by AlaaElattar**

### CODE REVIEW

**Comment by AlaaElattar** [comment]
Full VM
Screenshot from 2023-06-21 15-06-10
Screenshot from 2023-06-21 15-06-22
https://github.com/threefoldtech/tfgrid-sdk-ts/issues/557

**Comment by AlaaElattar** [comment]
Screenshot from 2023-06-25 13-13-09
Screenshot from 2023-06-25 13-13-13

Screenshot from 2023-06-25 13-13-23
https://github.com/threefoldtech/tfgrid-sdk-ts/issues/671
https://github.com/threefoldtech/tfgrid-sdk-ts/issues/557

File: packages/playground/src/components/select_solution_flavor_ssd.vue
Comment (MohamedElmdary) [inline packages/playground/src/components/select_solution_flavor_ssd.vue:1]: Why did we create this file?

**Comment by AhmedHanafy725** [comment]
please resolve the conflicts

**Comment by AhmedHanafy725** [comment]
the workflows are failing, can you check?

**Comment by AlaaElattar** [comment]
> the workflows are failing, can you check?

File: packages/playground/src/components/select_solution_flavor.vue [hunk 1]
```diff
@@ -66,11 +66,11 @@ import type { solutionFlavor } from "../types";
 type Package = PropType<solutionFlavor>;
 
 const props = defineProps({
-  minimum: { type: Object as Package, default: () => ({ cpu: 1, memory: 1024 * 2, disk: 15 }) },
-  standard: { type: Object as Package, default: () => ({ cpu: 2, memory: 1024 * 2, disk: 100 }) },
+  minimum: { type: Object as Package, default: () => ({ cpu: 1, memory: 1024 * 2, ssd: 15 }) },
+  standard: { type: Object as Package, default: () => ({ cpu: 2, memory: 1024 * 2, ssd: 100 }) },
```
Comment (MohamedElmdary) [inline packages/playground/src/components/select_solution_flavor.vue:70]:
```suggestion
  minimum: { type: Object as Package, default: () => ({ cpu: 1, memory: 1024 * 2, disk: 15 }) },
  standard: { type: Object as Package, default: () => ({ cpu: 2, memory: 1024 * 2, disk: 100 }) },
```

Why did we update `disk` to `ssd` here?

File: packages/playground/src/components/select_solution_flavor.vue [hunk 2]
```diff
@@ -66,11 +66,11 @@ import type { solutionFlavor } from "../types";
...
   recommended: {
     type: Object as Package,
-    default: () => ({ cpu: 4, memory: 1024 * 4, disk: 250 }),
+    default: () => ({ cpu: 4, memory: 1024 * 4, ssd: 250 }),
```
Comment (MohamedElmdary) [inline packages/playground/src/components/select_solution_flavor.vue:73]:
```suggestion
    default: () => ({ cpu: 4, memory: 1024 * 4, disk: 250 }),
```
Same here

File: packages/playground/src/components/select_solution_flavor.vue (hunk 1 above)
Comment (AlaaElattar) [reply packages/playground/src/components/select_solution_flavor.vue:70]: It was thabet's request. @xmonader

Comment (xmonader) [reply packages/playground/src/components/select_solution_flavor.vue:70]: it can be internally defined as disk, but to the user it should be shown as SSD Storage or SSD Disk

File: packages/playground/src/weblets/micro_vm.vue [hunk 3]
```diff
@@ -40,35 +40,12 @@
...
-        >
-          <input-tooltip tooltip="The amount of RAM (Random Access Memory) allocated to your instance.">
-            <v-text-field label="Memory (MB)" type="number" v-model.number="memory" v-bind="props" />
-          </input-tooltip>
-        </input-validator>
+        <SelectSolutionFlavor
+          :minimum="{ cpu: 1, memory: 1024 * 1, disk: 25 }"
+          :standard="{ cpu: 2, memory: 1024 * 4, disk: 50 }"
+          :recommended="{ cpu: 4, memory: 1024 * 4, disk: 100 }"
+          v-model="solution"
```
Comment (0oM4R) [inline packages/playground/src/weblets/micro_vm.vue:47]: please update the `weblet-layout` props with the solution
no need to use the cpu, memory, and disk refs now,

at the top of the weblet update the props to be like this code from dicourse weblet
```
<template>
  <weblet-layout
    ref="layout"
    :cpu="solution?.cpu"
    :memory="solution?.memory"
    :disk="(solution?.disk ?? 0) + rootFs(solution?.cpu ?? 0, solution?.memory ?? 0)"
    title-image="images/icons/discourse.png"
  >
```

File: packages/playground/src/weblets/full_vm.vue [hunk 4]
```diff
@@ -42,53 +42,12 @@
...
-            tooltip="The storage capacity allocated to your instance, indicating the amount of space available to store files, data, and applications."
-          >
-            <v-text-field label="Disk Size (GB)" type="number" v-model.number="diskSize" v-bind="props" />
-          </input-tooltip>
-        </input-validator>
+        <SelectSolutionFlavor
+          :minimum="{ cpu: 1, memory: 1024 * 1, disk: 25 }"
+          :standard="{ cpu: 2, memory: 1024 * 4, disk: 50 }"
+          :recommended="{ cpu: 4, memory: 1024 * 4, disk: 100 }"
+          v-model="solution"
```
Comment (0oM4R) [inline packages/playground/src/weblets/full_vm.vue:49]: same as I mentioned in micro vm

**Review by 0oM4R — CHANGES_REQUESTED** [review]
the filters do not update with the package values

[Screencast from 20 يول, 2023 EEST 01:17:15 م.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/f4136f2e-703c-4d01-b46c-4ccaae05ce82)

File: packages/playground/src/weblets/micro_vm.vue [hunk 5]
```diff
@@ -2,8 +2,8 @@
   <weblet-layout
     ref="layout"
     @mount="layoutMount"
-    :cpu="cpu"
-    :memory="memory"
+    :cpu="solution?.cpu"
+    :memory="solution?.memory"
     :disk="disks.reduce((total, disk) => total + disk.size, rootFsSize)"
```
Comment (0oM4R) [inline packages/playground/src/weblets/micro_vm.vue:7]: update the disk here to be consistence with the solution disk :
after logging  the filters i see the value is only 2 <the init value of `rootFsSize`
image
Screenshot from 2023-07-24 18-08-58

File: packages/playground/src/weblets/full_vm.vue [hunk 6]
```diff
@@ -228,7 +228,7 @@ async function deploy() {
           farmId: farm.value.farmID,
           farmName: farm.value.name,
           country: farm.value.country,
-          disks: [{ size: solution.value.disk, mountPoint: "/" }, ...disks.value],
+          disks: disks.value,
```
Comment (0oM4R) [inline packages/playground/src/weblets/full_vm.vue:231]: why did you remove the solution disk here ?

**Review by 0oM4R — APPROVED** [review]

**Review by AhmedHanafy725 — APPROVED** [review]
